### PR TITLE
Add libunifex/cci.20220430

### PR DESCRIPTION
**Pull request by Minimonium**
Specify library name and version:  **libunifex/cci.20220430**

A prototype library for the novel Sender/Receiver concurrency design based on [std::execution](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2300r5.html)

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libunifex/config.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+versions:
+  "2022.04.30":
```
Comment (uilianries) [inline recipes/libunifex/config.yml:2]:
```suggestion
  "cci.2022.04.30":
```
https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#what-version-should-packages-use-for-libraries-without-official-releases

**Review by uilianries — COMMENTED** [review]
Only need to update the version name and we are good to go.

File: recipes/libunifex/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    def package(self):
+        self.copy("LICENSE.txt", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "unifex")
+        self.cpp_info.set_property("cmake_target_name", "unifex::unifex")
```
Comment (prince-chrismc) [inline recipes/libunifex/all/conanfile.py:92]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "unifex")
        self.cpp_info.set_property("cmake_target_name", "unifex::unifex")
        self.cpp_info.set_property("pkg_config_name", "unifex")
```

File: recipes/libunifex/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        self.cpp_info.filenames["cmake_find_package"] = "unifex"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "unifex"
+        self.cpp_info.names["cmake_find_package"] = "unifex"
+        self.cpp_info.names["cmake_find_package_multi"] = "unifex"
```
Comment (prince-chrismc) [inline recipes/libunifex/all/conanfile.py:98]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "unifex"
        self.cpp_info.names["cmake_find_package_multi"] = "unifex"
        self.cpp_info.names["pkg_config"] = "unifex"
```

**Review by prince-chrismc — COMMENTED** [review]
just missing the pkg config names https://github.com/facebookexperimental/libunifex/blob/c359fd8e7d97d91359cf4a6c1dbef99b0b1767b6/source/CMakeLists.txt#L97

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
CI looks stuck 🤔 

You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

**Comment by uilianries** [comment]
> CI looks stuck 🤔
> 
> You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

Qt build eating all resources!

**Review by uilianries — APPROVED** [review]

**Pull request merged**